enhancements/machine-api/machine-health-checking.md

```diff
@@ -249,3 +263,175 @@ I want automated remediation on some nodes, but not all -- I may have "pet" mach
+This means that Machines must have all of the labels specified to be considered to match the selector.
+This could be used to specify that the MHC selects all Machines with the `cluster-api-machine-role: worker` label,
+which would match all MachineSets created by the installer, but not Control-Plane nodes,
+nor necessarily any MachineSets created by users post installation.
+
+#### Using expressions
+
+Using the `matchExpressions` part of the selector allows more complicated expressions to be created for selecting on labels.
+Each expression consists of a key, operator and values.
+This allows users to select a label with value that matches a set of values (operator: In),
+a label with value that is not in a set of values (operator: NotIn). Or check that a label exists or not.
+
+These expressions could be used to select all Machines that do not have the label `cluster-api-machine-role: master`,
+ie, all Machines in the cluster that are not part of the Control-Plane.
+This would include all MachineSets created by the installer and any created by users post installation.
+
+### How does the ClusterAutoscaler interact with MachineHealthCheck?
+
+The ClusterAutoscaler uses its own health checking to determine if a MachineSet is unhealthy after scaling up.
+Should it determine the MachineSet to be unhealthy,
+it would scale the MachineSet back down and try another MachineSet to add new capacity to the cluster.
+
+MachineHealthChecks could interfere with this health checking logic and prevent the autoscaler from adding new compute capacity.
+To ensure there are no conflicts, several scenarios must be tested to check whether there may be conflicts
+
+#### Scenarios
+##### Machine fails to launch due to user config error
+
+Steps to reproduce:
+- Create cluster on AWS
+- Deploy ClusterAutoscaler to cluster
+- Deploy MachineHealthCheck to cluster
+- Modify MachineSet to start on Spot instances with very low spot price
+  ```yaml
+  providerSpec:
+    value:
+      ...
+      spotMarketOptions:
+        maxPrice: "0.001" # Low to prevent EC2 instance being created
+  ```
+- Create deployment of dummy pods with some resources requested
+- Scale deployment until ClusterAutoscaler scales modified MachineSet
+
+What happens after Machine Fails:
+- If the number of Failed Machines is greater than `maxUnhealthy`:
+  - Example:
+    - This may happen if a large number of nodes are scaled in a single MachineSet and they all fail to provision
+      (eg AMI missing, or instance type not available in AZ)
+	- MachineHealthCheck:
+    - Remediation is short circuited, takes not action
+  - Cluster Autoscaler:
+    - Autoscaler determines nodes are unregistered
+    - After 15 minutes (MaxNodeProvisionTime) autoscaler scales MachineSets back as expected and re-attempts scaling
+	- Conclusion:
+    - In this scenario, the MHC has no effect on the operation of the Cluster Autoscaler
```

Comment:
should this line be indented to match the previous level? or maybe the "Conclusion" line should be pulled out?

(just curious because of the colon)

Reply:
Somehow mixed a tab in with the double spaces, odd 😅